### ISSUE TITLE
Response.IsSuccess proposal

### ISSUE THREAD

**Issue by mxpv** [go-resty/resty#169]
Hey.

I have an API which returns body in case of success or 4xx/5xx and error status text in case of error.
It would be nice to add `IsSuccess` helper, something like:
```
	if resp.IsSuccess() {
		return nil, errors.New(resp.Status())
	} else {
		return resp Body
	}
```

This is very similar `IsSuccessStatusCode` in https://stackoverflow.com/questions/32569860/checking-if-httpstatuscode-represents-success-or-failure

The implementation is trivial:
```
return statusCode >= 200 && statusCode <= 299;
```

What do you think?
Thanks.

**Comment by jeevatkm** [comment]
@mxpv I understand your recommendation. 

However, I would like to clarify about resty existing capabilities.

Resty has builtin support JSON and XML response handling(automatic unmarshal) for Success and Error scenario based on status code (Like you have described in the recommendation).

Let's say, if you have response struct for Success and Error.

For example:

```go
resp, err := resty.R().
		SetHeader(hdrContentTypeKey, jsonContentType).
		SetBody(user).
		SetResult(AuthSuccess{}).
		SetError(AuthError{}).
		Post("http://localhost:8080/login")

// Success
if resp.Result() != nil {
      result := resp.Result().(*AuthSuccess)
}

// Error
if resp.Error() != nil {
     error := resp.Error().(*AuthError)
}
``` 

Does it satisfy your use case?

**Comment by mxpv** [comment]
Not really. 
I don't have an error body, backend sends error description as http status, something like:
```
HTTP/1.0 400 login or password field empty!
Date: Sat, 28 Jul 2018 17:56:19 GMT
Content-Type: application/json
```

So right now I have something like this:
```
func (c Client) checkErr(resp *resty.Response) error {
	code := resp.StatusCode()
	if code >= 200 && code <= 299 {
		return nil
	}

	return &ProvError{Code: code, Message: resp.Status()}
}
```

**Comment by jeevatkm** [comment]
Okay, I will add `IsSuccess` method into `Response`.

FYI `SetResult` and `SetError` can be used separately.

**Comment by jeevatkm** [comment]
@mxpv helper methods added to master branch. Will be released in `v1.8.0`.

**Comment by mxpv** [comment]
@jeevatkm thanks a lot!
